flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/source/reader/CoordinatedSourceRescaleITCase.java

```diff
@@ -92,7 +92,7 @@ private void resumeCheckpoint(File checkpointDir, File restoreCheckpoint, int p)
             env.execute("resume checkpoint");
             throw new AssertionError("No success error");
         } catch (Exception e) {
-            if (RESTORED_CHECKPOINT != ExceptionUtils.getRootCause(e).getMessage()) {
+            if (!findThrowableWithMessage(e, RESTORED_CHECKPOINT).isPresent()) {
                 throw new AssertionError("Cannot resume", e);
             }
```

Comment:
Could we rephrase this like `assertThat(e, FlinkMatchers.containsMessage(CREATED_CHECKPOINT));`?